bob/src/cluster/quorum.rs

```diff
@@ -30,27 +30,34 @@ impl Quorum {
         debug!("PUT[{}] ~~~PUT LOCAL NODE FIRST~~~", key);
         let mut local_put_ok = 0_usize;
         let mut remote_ok_count = 0_usize;
-        let mut at_least = self.quorum;
+        let at_least = self.quorum;
         let mut failed_nodes = Vec::new();
         let (vdisk_id, disk_path) = self.mapper.get_operation(key);
-        if let Some(path) = disk_path {
+        let local_put_task = disk_path.map(|path| {
             debug!("disk path is present, try put local");
-            let res = put_local_node(&self.backend, key, data, vdisk_id, path).await;
-            if let Err(e) = res {
+            put_local_node(&self.backend, key, data, vdisk_id, path).boxed()
+        });
+        debug!(
+            "PUT[{}] need at least {} additional puts",
+            key,
+            at_least - 1
+        );
+        debug!("PUT[{}] ~~~PUT TO REMOTE NODES~~~", key);
+        let (mut tasks, mut errors) = self.put_remote_nodes(key, data, at_least - 1).await;
+        if let Some(task) = local_put_task {
+            let local_put = task.await;
+            if let Err(e) = local_put {
                 error!("{}", e);
                 failed_nodes.push(self.mapper.local_node_name().to_owned());
+                debug!("PUT[{}] local failed, put another remote", key);
             } else {
                 local_put_ok += 1;
-                at_least -= 1;
                 debug!("PUT[{}] local node put successful", key);
             }
-        } else {
-            debug!("skip local put");
+        };
+        if local_put_ok == 0 {
```

Comment:
This should be run only if local put finished with error. It should not if local put wasn't started at all (because there is no need to put to the local node)